src/lib/actions/AbstractRunAction.ts

```diff
@@ -76,16 +78,27 @@ export abstract class AbstractRunAction implements Action {
 		const targetPaths: string[] = this.inputProcessor.resolveTargetPaths(inputs);
 		const runOptions: RunOptions = this.inputProcessor.createRunOptions(inputs, this.isDfa());
 		const engineOptions: EngineOptions = this.engineOptionsFactory.createEngineOptions(inputs);
+
 		const outputOptions: RunOutputOptions = this.inputProcessor.createRunOutputOptions(inputs);
+		const verboseViolations: boolean = inputs["verbose-violations"] as boolean;
+
+		const runResultsProcessor: RunResultsProcessor = new RunResultsProcessor(this.display, outputOptions, verboseViolations);
+		const resultsProcessors: ResultsProcessor[] = [runResultsProcessor];
+		const internalOutfile: string = process.env[ENV_VAR_NAMES.SCANNER_INTERNAL_OUTFILE];
+		if (internalOutfile && internalOutfile.length > 0) {
+			const internalOutputFormat: OutputFormat = inferFormatFromInternalOutfile(internalOutfile);
+			resultsProcessors.push(new OutfileResultsProcessor(internalOutputFormat, internalOutfile, verboseViolations));
+		}
+		const compositeResultsProcessor: ResultsProcessor = new CompositeResultsProcessor(resultsProcessors);
```

Comment:
Eventually we should probably separate the RunResultsProcessor into 3 separate ResultsProcessor instances (1 to create the json output, 1 to write to console, and 1 to write to file) and then move all of the construction of these into a ResultsProcessorFactory that we can inject as a dependency to this AbstractRunAction.

But it was a bit out of the scope of this work item. We'll see how long this v3 code lives... if we evolve it into v4 then we are close to having the ideal open/closed principle adherence with our input/output handling.